Update serial communication and clean up duplicated & outdated info (#1317)
ARMmbed/Handbook: docs/quick-start/quick-start-cli-debug.md

## Using printf

> The easiest way to do basic debugging is to use the `printf` command in your code, then read the output using a serial terminal.

Reviewer: ```suggestion
The easiest way to do basic debugging is to use the `printf` function in your code, then read the output using a serial terminal.
```

> For example, add `printf("Hello World!\n\r");` to the top of your main function, and then recompile the program and flash it to your device. To view the output, open a serial terminal:

Reviewer: ```suggestion
For example, add `printf("Hello World!\n");` to the top of your main function, and then recompile the program and flash it to your device. To view the output, open a serial terminal:
```
`\r` is no longer required.

```
mbed sterm
```

> You may need to reset the board, because messages are visible if printed while the terminal is open.

Reviewer 2: ```suggestion
You may need to reset the board, because messages are visible only if printed while the terminal is open.
```

> To close the serial terminal, press Ctrl + C.

Reviewer 2: ```suggestion
To close the serial terminal, enter <kbd>Ctrl + C</kbd>.
```

<span class="notes">**Note:** Unless otherwise specified, `printf` defaults to a baud rate of `9600` on Mbed OS. You can modify this value in the `mbed_app.json` file. To configure your terminal client to this baud rate, change the speed option when selecting the port. You can view the [configuration options page](../program-setup/advanced-configuration.html) to learn more about how to configure OS-level options.</span>

> For more options of `mbed sterm`, see [Board to PC communication over USB](../tutorials/serial-comm.html).

Reviewer 2: ```suggestion
For more `mbed sterm` options, see [Board to PC communication over USB](../tutorials/serial-comm.html).
```
